app/assets/javascripts/dashboard.js

```diff
@@ -112,7 +112,7 @@ MinionPoller = {
         handleBootstrapButtonTitle();
 
         // show/hide "update all nodes" link
-        $("#update-all-nodes").toggleClass('hidden', !(updateAvailable && allApplied));
+        $("#update-all-nodes").toggleClass('hidden', !updateAvailable || !allApplied);
```

Comment:
I **think** that this happening due to `!updateAvailable === !true === false` but with the `||` !allApplied === !false === true` means the button is hidden.